### PR TITLE
Upgrade elm to 0.18, some refactoring

### PR DESCRIPTION
**Pull request by magopian**
What do you think @n1k0 @glasserc @natim?

The big piece was coping with the API change between evancz/elm-http and
elm-lang/http (I like the new API a lot ;)

As a reminder: for the moment, only the GET APIs are included in the
"kinto.elm" library, so some "old" code is still around (for adding/editing).

### CODE REVIEW

File: src/Kinto.elm [hunk 1]
```diff
@@ -201,44 +199,50 @@ configure baseUrl auth =
...
+
+
+toKintoResponse :
+    (Result Error Json.Value -> msg)
+    -> Result Http.Error Json.Value
+    -> msg
+toKintoResponse toMsg response =
+    response
+        |> Result.mapError extractError
+        |> toMsg
```
Comment (glasserc) [inline src/Kinto.elm]: What's the value of having `toMsg` as a parameter instead of just calling it afterwards?

File: src/Model.elm [hunk 2]
```diff
@@ -108,49 +94,57 @@ update msg model =
...
+                              }
+                            , Cmd.none
+                            )
+
+                Err error ->
+                    let
+                        _ =
+                            Debug.log "FetchRecordResponse failed:" error
+                    in
+                        ( model, Cmd.none )
```
Comment (glasserc) [inline src/Model.elm]: Why not update the `error` field here, as you do above?

**Review by glasserc — COMMENTED** [review]
I didn't stare at it extensively, but at a quick glance, it seems OK.

File: src/Kinto.elm (hunk 1 above)
Comment (magopian) [reply src/Kinto.elm]: well spotted, didn't properly complete my refactorization ;)

File: src/Model.elm (hunk 2 above)
Comment (magopian) [reply src/Model.elm]: eh, makes sense, thanks!

**Comment by magopian** [comment]
Thanks @glasserc for the comments, I believe I addressed them in 8e67cff

**Pull request merged**